integration/pod_test.go

```diff
@@ -0,0 +1,180 @@
+//go:build integration
+// +build integration
+
+package integration
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/namespace"
+	"github.com/openshift-kni/eco-goinfra/pkg/pod"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func TestPodCreate(t *testing.T) {
+	t.Parallel()
+	client := clients.New("")
+	assert.NotNil(t, client)
+
+	const (
+		testNamespace = "egi-pod-create-ns"
+		podName       = "create-test"
+	)
+
+	// Create a namespace in the cluster using the namespaces package
+	namespaceBuilder := namespace.NewBuilder(client, testNamespace)
+	assert.NotNil(t, namespaceBuilder)
+
+	// Preemptively delete the namespace before the test
+	err := namespaceBuilder.DeleteAndWait(time.Duration(30) * time.Second)
+	assert.Nil(t, err)
+
+	// Create the namespace
+	namespaceBuilder, err = namespaceBuilder.Create()
+	assert.Nil(t, err)
+
+	// Defer the deletion of the namespace
+	defer func() {
+		// Delete the namespace
+		err := namespaceBuilder.Delete()
+		assert.Nil(t, err)
+	}()
+
+	testContainerBuilder := pod.NewContainerBuilder("test", containerImage, []string{"sleep", "3600"}).
+		WithSecurityContext(&corev1.SecurityContext{
+			RunAsUser:  nil,
+			RunAsGroup: nil,
+		})
+	containerDefinition, err := testContainerBuilder.GetContainerCfg()
+	assert.Nil(t, err)
+
+	podBuilder := pod.NewBuilder(client, podName, testNamespace, containerImage)
+	podBuilder = podBuilder.RedefineDefaultContainer(*containerDefinition)
+
+	// Create a pod in the namespace
+	_, err = podBuilder.CreateAndWaitUntilRunning(timeoutDuration)
+	assert.Nil(t, err)
+
+	// Check if the pod was created
+	podBuilder, err = pod.Pull(client, podName, testNamespace)
+	assert.Nil(t, err)
+	assert.NotNil(t, podBuilder.Object)
+}
+
+func TestPodDelete(t *testing.T) {
+	t.Parallel()
+	client := clients.New("")
+	assert.NotNil(t, client)
+
+	const (
+		testNamespace = "egi-pod-delete-ns"
+		podName       = "delete-test"
+	)
+
+	// Create a namespace in the cluster using the namespaces package
+	namespaceBuilder := namespace.NewBuilder(client, testNamespace)
+	assert.NotNil(t, namespaceBuilder)
+
+	// Preemptively delete the namespace before the test
+	err := namespaceBuilder.DeleteAndWait(time.Duration(30) * time.Second)
+	assert.Nil(t, err)
+
+	// Create the namespace
+	namespaceBuilder, err = namespaceBuilder.Create()
+	assert.Nil(t, err)
```

Comment:
We can replace this with: https://github.com/openshift-kni/eco-goinfra/blob/main/integration/common.go#L13